cmd/system-probe/config/config.go

```diff
@@ -0,0 +1,193 @@
+		}
+	} else {
+		log.Infof("no config exists at %s, ignoring...", configPath)
+	}
+
+	return load(configPath)
+}
+
+func load(configPath string) (*Config, error) {
+	cfg := aconfig.Datadog
+
+	if err := aconfig.ResolveSecrets(cfg, filepath.Base(configPath)); err != nil {
+		return nil, err
+	}
+
+	c := &Config{
+		Enabled:             cfg.GetBool(key(spNS, "enabled")),
+		EnabledModules:      make(map[ModuleName]struct{}),
+		ExternalSystemProbe: cfg.GetBool(key(spNS, "external")),
+
+		SocketAddress:      cfg.GetString(key(spNS, "sysprobe_socket")),
+		MaxConnsPerMessage: cfg.GetInt(key(spNS, "max_conns_per_message")),
+
+		LogFile:   cfg.GetString(key(spNS, "log_file")),
+		LogLevel:  cfg.GetString(key(spNS, "log_level")),
+		DebugPort: cfg.GetInt(key(spNS, "debug_port")),
+
+		ProfilingEnabled:     cfg.GetBool(key(spNS, "profiling.enabled")),
+		ProfilingSite:        cfg.GetString(key(spNS, "profiling.site")),
+		ProfilingURL:         cfg.GetString(key(spNS, "profiling.profile_dd_url")),
+		ProfilingAPIKey:      aconfig.SanitizeAPIKey(cfg.GetString(key(spNS, "profiling.api_key"))),
+		ProfilingEnvironment: cfg.GetString(key(spNS, "profiling.env")),
+	}
+
+	if err := ValidateSocketAddress(c.SocketAddress); err != nil {
+		log.Errorf("Could not parse %s.sysprobe_socket: %s", spNS, err)
+		c.SocketAddress = defaultSystemProbeAddress
+		cfg.Set(key(spNS, "sysprobe_socket"), c.SocketAddress)
+	}
+
+	if c.MaxConnsPerMessage > maxConnsMessageBatchSize {
+		log.Warn("Overriding the configured connections count per message limit because it exceeds maximum")
+		c.MaxConnsPerMessage = defaultConnsMessageBatchSize
+		cfg.Set(key(spNS, "max_conns_per_message"), c.MaxConnsPerMessage)
+	}
+
+	// this check must come first so we can accurately tell if system_probe was explicitly enabled
+	if cfg.GetBool("network_config.enabled") {
+		log.Info("network_config.enabled detected: enabling system-probe with network module running.")
+		c.EnabledModules[NetworkTracerModule] = struct{}{}
+	} else if c.Enabled && !cfg.IsSet("network_config.enabled") {
+		// This case exists to preserve backwards compatibility. If system_probe_config.enabled is explicitly set to true, and there is no network_config block,
+		// enable the connections/network check.
+		log.Info("network_config not found, but system-probe was enabled, enabling network module by default")
+		c.EnabledModules[NetworkTracerModule] = struct{}{}
+	}
+
+	if cfg.GetBool(key(spNS, "enable_tcp_queue_length")) {
+		log.Info("system_probe_config.enable_tcp_queue_length detected, will enable system-probe with TCP queue length check")
+		c.EnabledModules[TCPQueueLengthTracerModule] = struct{}{}
+	}
+	if cfg.GetBool(key(spNS, "enable_oom_kill")) {
+		log.Info("system_probe_config.enable_oom_kill detected, will enable system-probe with OOM Kill check")
+		c.EnabledModules[OOMKillProbeModule] = struct{}{}
+	}
+	if cfg.GetBool("runtime_security_config.enabled") || cfg.GetBool("runtime_security_config.fim_enabled") {
```

Comment:
I would move this logic in each module, it will decouple the modules a bit more. What about adding a `IsEnabled(*Config) bool` function in the `Factory` struct ? Probably something we could do in another PR.

Reply:
That would create an import loop, because then the config would import the modules, which import the config.